propagation/baggage.go

```diff
@@ -29,6 +29,19 @@ func (b Baggage) Inject(ctx context.Context, carrier TextMapCarrier) {
 
 // Extract returns a copy of parent with the baggage from the carrier added.
```

Comment:
I think in the Go Doc comment we could add something like 

```
If carrier implements [MultiGetter] (e.g. [HeaderCarrier]), Values is invoked for multiple values extraction. Otherwise, Get is called.
```

Inspired by https://pkg.go.dev/io#WriteString